dotbot/frontend/src/DotBots.js

```diff
@@ -15,6 +15,47 @@ const DotBots = ({ dotbots, updateDotbots, publishCommand, publish }) => {
   const [ showDotBotHistory, setShowDotBotHistory ] = useState(true);
   const [ dotbotHistorySize, setDotbotHistorySize ] = useState(maxPositionHistory);
   const [ showSailBotHistory, setShowSailBotHistory ] = useState(true);
+  const dotbotsRef = useRef(dotbots);
+  useEffect(() => {
+    dotbotsRef.current = dotbots;
+  }, [dotbots]);
+
+  const runAlgorithm = useCallback(async () => {
+    const currentDotbots = dotbotsRef.current;
+
+    const botRadius = 0.02;
+
+    // Create all agents list
+    const agents = [];
```

Comment:
I don't understand why this code is here. It doesn't seem to be used or I am missing something